### ISSUE TITLE
Adding Lags to dynamic features

### ISSUE THREAD

**Issue by pst2154** [Nixtla/mlforecast#124]
## Summary
- Add lags to dynamic regressors

## Motivation
- I wish to include historical values of dynamic variables

## Description
- currently I have to define a fake forecast function, then preprocess, then change the names, then make a model forecast function

## Example
```
def add_lags(df,lags,var, name_conv):
    fcst_data = MLForecast(
        models=[],
        freq='W',
        lags=[i+1 for i in range(lags)]
    )
    new_full_df = fcst_data.preprocess(data=df, id_col='unique_id',time_col='ds', target_col=var)    
    new_names = list(set(new_full_df.columns) - set(full_df.columns))
    for name in new_names:
        new_full_df.rename(columns={name:name_conv + name},inplace = True)
    
    return new_full_df
```

**Comment by jmoralez** [comment]
Thanks for raising this. To be honest that's not a bad solution, that allows you to compute transformations on the lags as well. I'll try to think of an interface for this.

Do you then split the future values of your dynamic features for the predict step?

**Comment by pst2154** [comment]
Yes, i computed all features then did a split then fit forecast model then used dynamic features to forecast, it’s a bit tedious but it works ok. I’ll see if the CV functions can work out of the box though

**Comment by jmoralez** [comment]
The cross_validation method of MLForecast method computes the dynamic_dfs internally, so it should work. LightGBMCV doesn't do that yet, I'll work on that.

**Comment by jmoralez** [comment]
v0.7.1 added that functionality to LightGBMCV, you should be able to use it by upgrading the library.

**Comment by pst2154** [comment]
I have not tested 0.7.1 but I got something to run last night prior to you adding a fix: It seemed to do what I wanted it to do. What fix did you make?

```
cv = LightGBMCV(
    freq='W',
    lags=[],
    lag_transforms={},
    num_threads=1
)
params = {
    'bagging_freq': 1,
    'learning_rate': 0.05,
    'verbose': -1,
}
dyn = filter_train[cols + ['ds','unique_id','y']]
hist = cv.fit(
    filter_train[cols + ['ds','unique_id','y']], static_features=['feat1', 'feat2'],
    n_windows=5,
    window_size=26,
    step_size=26,
    input_size = 30,
    compute_cv_preds=True,
    params=params,
    dynamic_dfs=[dyn],
    metric = 'rmse'
)
```

**Comment by jmoralez** [comment]
Oh it just removes the `dynamic_dfs` argument, since all future dynamic values are known in each fold (they can be extracted from the validation set).
